Modules/timemodule.c

```diff
@@ -2088,7 +2088,7 @@ pysleep(_PyTime_t secs)
         err = ret;
 #else
         Py_BEGIN_ALLOW_THREADS
-        ret = select(0, (fd_set *)0, (fd_set *)0, (fd_set *)0, &timeout);
+        ret = nanosleep(&timeout, NULL);
```

Comment:
For maximum portability, I would prefer to keep select() for platforms which don't support nanosleep(). Could you add a test in configure.ac and add the `#ifdef`  to have 3 code paths (select, nanosleep, clock_nanosleep)?